vault/audit.go

```diff
@@ -90,6 +90,12 @@ func (c *Core) enableAudit(ctx context.Context, entry *MountEntry) error {
 	viewPath := auditBarrierPrefix + entry.UUID + "/"
 	view := NewBarrierView(c.barrier, viewPath)
 
+	// Mark the view as read-only until the mounting is complete and
+	// ensure that it is reset after. This ensures that there will be no
+	// writes during the construction of the backend.
+	view.markReadOnly(true)
+	defer view.markReadOnly(false)
```

Comment:
This will have different behavior than the clode below in setupAudits. In that case, if we abort setting up the backend due to an error from newAuditBackend (or the backend being nil), we don't then mark it as read-write, whereas here we do. I'm not sure what the better solution should be (probably to keep it read-only) but I think we should be consistent.

Reply:
Now marking it as read-write in all the exit scenarios.